### ISSUE TITLE
Feature Request: generator function should be passed with typedef

### ISSUE THREAD

**Issue by zzmingo** [palantir/tslint#2734]
### Feature Request

- __TSLint version__: 5.2.0
- __TypeScript version__: 2.3.1
- __Running TSLint via__: Visual Studio

#### tsconfig.json

```json
"typedef": [
  true, 
  "call-signature", 
  "parameter", 
  "member-variable-declaration",
  "arrow-parameter",
  "property-declaration"
],
```

#### TypeScript code being linted

```ts
function* testGeneratorFunction() {
}

let iterator: Iterator<any> = testGeneratorFunction()
```

#### Actual behavior

> [tslint] expected call-signature: 'test' to have a typedef (typedef)

#### Expected behavior

tslint should pass because generator function always return an instance type of Iteractor<>

Maybe there is a option `disable-generator-function` for rule `typedef`

**Comment by andy-hanson** [comment]
`typedef` will require a type definition to exist even when it can be inferred. I don't think there should be an exception here.
